lib/core/lib/ops/userOps.js

```diff
@@ -20,6 +21,49 @@ function buildUserOps (db/*, config */) {
     job.client.query(sqlQuery, utils.boomErrorWrapper(next))
   }
 
+  function readUserTeams (job, next) {
+    let { id, organizationId, limit, page } = job
+    let sqlQuery = SQL`
+        SELECT
+          teams.id,
+          teams.name,
+          teams.description,
+          teams.path,
+          teams.org_id
+        FROM teams
+        LEFT JOIN team_members ON team_members.team_id = teams.id
+        WHERE team_members.user_id = ${id} AND teams.org_id = ${organizationId}
+        GROUP BY teams.id, teams.name, teams.description, teams.path, teams.org_id
+        ORDER BY UPPER(name)
+      `
+
+    if (limit) {
+      sqlQuery.append(SQL` LIMIT ${limit}`)
+    }
+    if (limit && page) {
+      let offset = (page - 1) * limit
+      sqlQuery.append(SQL` OFFSET ${offset}`)
+    }
+
+    job.client.query(sqlQuery, function (err, result) {
+      if (err) return next(Boom.badImplementation(err))
+
+      sqlQuery = SQL`
+            SELECT COUNT(*)::INTEGER AS cnt
+            FROM team_members
+            WHERE user_id = ${id}
+          `
+      job.client.query(sqlQuery, function (err, countResult) {
+        if (err) return next(Boom.badImplementation(err))
+
+        job.totalTeamsCount = countResult.rowCount > 0 ? parseInt(countResult.rows[0].cnt, 10) : 0
+        job.user.teamsCount = result.rowCount
```

Comment:
It would be nicer if you set the properties right after you have the result. It's much more readable. I don't think it makes a diference if you set them in sequence rather than all at once because if an error appears the already set values aren't used anyway.

Reply:
nicer but no point to do assignments if not sure that it executes till end